cetmix_tower_server/models/cx_tower_variable_value.py

```diff
@@ -92,6 +102,32 @@ class TowerVariableValue(models.Model):
         ),
     ]
 
+    @api.depends("option_id", "variable_id.option_ids")
+    def _compute_value_char(self):
+        """
+        Compute the 'value_char' field, which holds the string representation
+        of the selected option for the variable.
+        """
+        for rec in self:
+            if rec.variable_id.option_ids and rec.option_id:
+                rec.value_char = rec.option_id.name
+            elif not rec.variable_id.option_ids:
+                rec.value_char = rec.value_char or ""
+
+    @api.onchange("variable_id")
+    def _onchange_variable_id(self):
+        """
+        Reset option_id when variable changes or
+        doesn't have options
+        """
+        for rec in self:
+            if rec.variable_id.option_ids:
+                return {
+                    "domain": {"option_id": [("variable_id", "=", rec.variable_id.id)]}
```

Comment:
Return domain is deprecated and should not be used. However you can store domain in a binary field and apply it later. Please check the Dev Memo for details.